gamemodes/zombiesurvival/gamemode/sh_weaponquality.lua

```diff
@@ -135,6 +135,7 @@ function GM:CreateWeaponOfQuality(i, orig, quality, classname, branch)
 	-- TODO: Refactor me to not use the full weapon class once all self.BaseClass calls have been removed
 
 	local wept = weapons.Get(classname)
+	wept.ClassName = newclass
```

Comment:
what's this for?

Reply:
It seemed to fix the grey model and weapon quality issue, but after much more testing it also seems to have broken gunturret. You can still buy it, it'll be in your inventory, but you cannot deploy it. When you select the gunturret, it just flashed green and then the deploy thingy disappeared.

I removed it now. After that the grey models issue is back, along with the weapon qualities error:
[error.log](https://github.com/JetBoom/zombiesurvival/files/10694897/error.log)